src/docfx/build/page/BuildPage.cs

```diff
@@ -26,49 +26,60 @@ internal static class BuildPage
 
             var dependencies = new DependencyMapBuilder();
 
-            var (errors, pageType, content, fileMetadata) = await Load(file, dependencies, bookmarkValidator, buildChild);
-            var conceptual = content as Conceptual;
+            var (errors, schema, model) = await Load(file, dependencies, bookmarkValidator, buildChild);
 
-            var locale = file.Docset.Config.Locale;
-            var metadata = JsonUtility.Merge(Metadata.GetFromConfig(file), fileMetadata);
-            var docId = file.Docset.Redirections.TryGetDocumentId(file, out var id) ? id : file.Id;
+            model.PageType = schema.Name;
+            model.Locale = file.Docset.Config.Locale;
+            model.Metadata = JsonUtility.Merge(Metadata.GetFromConfig(file), model.Metadata);
+            model.ShowEdit = file.Docset.Config.Contribution.ShowEdit;
 
-            // TODO: add check before to avoid case failure
-            var (repoError, author, contributors, updatedAt) = await contribution.GetContributorInfo(file, metadata.Value<string>("author"));
-            if (repoError != null)
-                errors.Add(repoError);
-            var (editUrl, contentUrl, commitUrl) = contribution.GetGitUrls(file);
+            if (schema.Attribute is PageSchemaAttribute pageSchema)
+            {
+                if (pageSchema.DocumentId)
```

Comment:
>if (pageSchema.DocumentId) [](start = 16, length = 26)

How about putting this block  into a SetupModel() internal method for readability?

Reply:
actually it's not pretty and common, what if TOC also need this logic to control the output?

---
In reply to: [213583003](https://github.com/dotnet/docfx/pull/3316#discussion_r213583003) [](ancestors = 213583003)